e2e/test/ConnectionStatusChangeHandlerTests.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Diagnostics.Tracing;
+    using System.Net;
+    using System.Threading.Tasks;
+
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    public class ConnectionStatusChangeHandlerTests : IDisposable
+    {
+        private readonly string DevicePrefix = $"E2E_{nameof(ConnectionStatusChangeHandlerTests)}";
+        private readonly string ModulePrefix = $"E2E_{nameof(ConnectionStatusChangeHandlerTests)}";
+        private static TestLogging _log = TestLogging.GetInstance();
+
+        private readonly ConsoleEventListener _listener;
+
+        public ConnectionStatusChangeHandlerTests()
+        {
+            _listener = TestConfig.StartEventListener();
+        }
+
+        [TestMethod]
+        public async Task DeviceClient_DeviceDeleted_Gives_ConnectionStatus_DeviceDisabled()
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix + "_DeviceClient").ConfigureAwait(false);
+            string deviceConnectionString = testDevice.ConnectionString;
+
+            var config = new Configuration.IoTHub.DeviceConnectionStringParser(deviceConnectionString);
+            string iotHub = config.IoTHub;
+            string deviceId = config.DeviceID;
+            string key = config.SharedAccessKey;
+
+            SharedAccessSignatureBuilder builder = new SharedAccessSignatureBuilder()
+            {
+                Key = key,
+                TimeToLive = new TimeSpan(0, 10, 0),
+                Target = $"{iotHub}/devices/{WebUtility.UrlEncode(deviceId)}",
+            };
+
+            DeviceAuthenticationWithToken auth = new DeviceAuthenticationWithToken(deviceId, builder.ToSignature());
+
+            ConnectionStatus? status = null;
+            ConnectionStatusChangeReason? statusChangeReason = null;
+            bool deviceDisabledReceived = false;
+
+            using (DeviceClient deviceClient = DeviceClient.Create(iotHub, auth, Client.TransportType.Amqp_Tcp_Only))
```

Comment:
It's over complicated. Could be optimized to be `CreateFromConnectionString.CreateFromConnectionString(deviceConnectionString)`.